tensorflow/contrib/layers/python/layers/pooling.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2015 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+# pylint: disable=unused-import,g-bad-import-order
+"""Contains the pooling layer classes and their functional aliases.
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from tensorflow.python.framework import dtypes
+from tensorflow.python.framework import tensor_shape
+from tensorflow.python.layers import base
+from tensorflow.python.ops import array_ops
+from tensorflow.python.ops import math_ops
+
+
+
+
+def spatial_pyramid_pooling(inputs, spatial_bin_dimensions=None, mode='max'):
+  """Spatial pyramid pooling (SPP) is a pooling strategy to result in an output of fixed size.
+
+  Args:
+    inputs: The tensor over which to pool. Must have rank 4.
+    spatial_bin_dimensions: The list of bin sizes over which pooling is to be done.
+    mode: Pooling mode 'max' or 'avg'.
+
+  Returns:
+    Output tensor.
+  """
+  layer = SpatialPyramidPooling(spatial_bin_dimensions=spatial_bin_dimensions, mode=mode)
+  return layer.apply(inputs)
+
+
+class SpatialPyramidPooling(base.Layer):
+  """Spatial pyramid pooling (SPP) is a pooling strategy to result in an output of fixed size.
+
+    Arguments:
+        spatial_bin_dimensions: The list of dimensions define the output dimension
+          of each pooling level. The value of each dimenstion is the level of
+          the spatial pyramid.
```

Comment:
Sorry, this description is confusing, please clarify. Consider that most people reading this docstring come to it with no prior context.